modules/offers/server/controllers/offers.server.controller.js

```diff
@@ -608,6 +608,25 @@ exports.list = function (req, res) {
     }
   }
 
+  // Filter out users that do not share any circles with the authenticated user
+  // and chose to not appear in those searches.
+  const showOnlyInMyCirclesQueries = [
+    { showOnlyInMyCircles: false },
+    { showOnlyInMyCircles: { $exists: false } },
+  ];
+  req.user.member.forEach(function (membership) {
```

Comment:
We currently have about 13K members without `member` key in their profile — I can run query to add it to everyone so that this won't fail due to that inconsistency.

First it would be good to add `default: []` to here though:

https://github.com/Trustroots/trustroots/blob/5daa64f6af253ce4274bfdf5719cddf2b88cc109/modules/users/server/models/user.server.model.js#L337-L340

Reply:
I'll add the default. I don't know what is better to have, the `$exists` and the `req.user.member?.forEach(...)` or force-adding those attributes into the db? (note `req.user` is known to exist as we verify that at the very start of that function so we don't need the `?` behind `req.user`).

Reply:
`req.user` will always exist and thus `?` is needed only in `member?`, correct! :-)

Reply:
I'll add those both (`member` and `showOnlyInMyCircles`) attributes to the DB anyway, just because I want to have the data as predictable as possible. That said, it's good to be a little bit defensive in coding when it makes sense, as long as it's not to the point that it would hide genuine bugs.

Adding `$exists` in the query makes the query a tiny bit slower/more complex so I'd probably leave that out to keep things simple.

Reply:
How do we ensure that these attributes are also added to existing databases of developers?
Is there some sort of database update scripts?

Reply:
Here are the database migration scripts: https://github.com/Trustroots/trustroots/tree/dd7ee07ab92f466d41c15df8f6a8852cc1377e2b/migrations

But unlikely readme says, those don't run on `npm start` automatically anymore and need to be run manually.

I wouldn't worry too much about locale databases — it's quick to wipe out the database and re-populate it with new test data.